scripts/dts/extract/flash.py

```diff
@@ -91,6 +91,27 @@ def _extract_partition(self, node_path):
 
         insert_defs("DT_FLASH_AREA", prop_def, prop_alias)
 
+    def _extract_partition_dev(self, node_address):
+        # Extract a per partition dev name, something like:
+        # #define DT_FLASH_AREA_IMAGE_0_DEV="FLASH_CTRL"
+
+        # For now assume node_address is something like:
+        # /flash-controller@4001E000/flash@0/partitions/partition@fc000
+        # first we go up 2 levels to get the flash, check its compat
+        #
+        # The flash controller might be the flash itself (for cases like NOR
+        # flashes), for the case of 'soc-nv-flash' we assume its the parent
+        # of the flash node.
+        ctrl_addr = '/' + '/'.join(node_address.split('/')[1:-2])
+        if get_compat(ctrl_addr) == 'soc-nv-flash':
+            ctrl_addr = '/' + '/'.join(node_address.split('/')[1:-3])
```

Comment:
Could do this maybe. I didn't catch right away that the calculated value was overridden.

```python
if get_compat(ctrl_addr) == 'soc_nv_flash':
    ...
else:
    ...
```

Nice comment above by the way!